### DISCUSSION TITLE
Kekulization of dummy atoms next to aromatic system

### DISCUSSION THREAD

**Discussion by laurenreid1** [rdkit/rdkit#4716] [Q&A] [answered]
Hi everyone,

I'd like to depict a benzocyclohexane-like compound containing 2 dummy atoms within the cyclohexane ring (e.g c12ccccc1*-*CC2). Using the following code:

```
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import rdMolDraw2D

mol = Chem.MolFromSmiles("c12ccccc1*-*CC2")
dm = Draw.PrepareMolForDrawing(mol)
d2d = rdMolDraw2D.MolDraw2DCairo(240, 135)
d2d.DrawMolecule(dm)
d2d.FinishDrawing()
draw_text = d2d.GetDrawingText()

with open("test.png", "wb") as f:
    f.write(draw_text)

```

I receive this depicition that adds a double bond between the 2 dummies:
![image](https://user-images.githubusercontent.com/8626999/142643837-a73e4079-97fd-4648-a65b-b0d9b9b97397.png)

I believe this is happening during kekulization, where the conjugation of the dummy atoms to the benzene ring is interpreted as aromaticity? I can turn off the behaviour using the following:

```
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import rdMolDraw2D

mol = Chem.MolFromSmiles("c12ccccc1*-*CC2", sanitize=False)
mol.UpdatePropertyCache()
dm = Draw.PrepareMolForDrawing(mol, kekulize=False)
d2d = rdMolDraw2D.MolDraw2DCairo(240, 135)
d2d.drawOptions().prepareMolsBeforeDrawing = False
d2d.DrawMolecule(dm)
d2d.FinishDrawing()
draw_text = d2d.GetDrawingText()

with open("test.png", "wb") as f:
    f.write(draw_text)
```

But this leaves me with the following depiction:

![image](https://user-images.githubusercontent.com/8626999/142645386-6e3985a8-ef8a-4e77-977f-18408c0d1a00.png)

Does anyone know if there's a way I can depict the benzene ring in its kekulized form while keeping the bond between the dummy atoms as a single aliphatic bond? I've been playing around with the santizeflags but haven't found a combination that will allow only the benzene bonds to be kekulized.

Thanks,
Lauren

EDIT: My RDKit version is 2020.03.3

**Comment by ptosco** [comment, marked as answer]
@laurenreid1 That looks like a bug to me; actually that double bonds is not even marked as being conjugated.
For the time being, to fix the depiction you only need to explicitly set the `BondType` of the bond between the two dummies as `SINGLE`:

```
from rdkit import Chem
from rdkit.Chem.Draw import rdMolDraw2D
from IPython.display import Image

mol = Chem.MolFromSmiles("c12ccccc1*-*CC2")

dm = Draw.PrepareMolForDrawing(mol)
for b in dm.GetBonds():
    if b.GetBeginAtom().GetAtomicNum() == 0 and b.GetEndAtom().GetAtomicNum() == 0:
        b.SetBondType(Chem.BondType.SINGLE)
d2d = rdMolDraw2D.MolDraw2DCairo(240, 135)
d2d.drawOptions().prepareMolsBeforeDrawing = False
d2d.DrawMolecule(dm)
d2d.FinishDrawing()
draw_text = d2d.GetDrawingText()

Image(draw_text)
```
![image](https://user-images.githubusercontent.com/5244385/142769359-767d540c-d782-406d-931c-ac921acc10ab.png)

**Reply by laurenreid1** [reply]
Hi Paolo,

Thank you for the suggestion. That works nicely.

Lauren
